CMakeLists.txt

```diff
@@ -195,72 +195,60 @@ target_link_libraries(
         numa
 )
 
-# Note on flags:
-#   ttnn and tt_lib will break if built with LTO, so leaving -fno-lto in compile options
-add_library(linker_flags INTERFACE)
-
-add_library(compiler_warnings INTERFACE)
-target_compile_options(
-    compiler_warnings
-    INTERFACE
-        -Werror
-        -Wdelete-non-virtual-dtor
-        -Wreturn-type
-        -Wswitch
-        -Wuninitialized
-        -Wno-unused-parameter
-)
-
-# add additional compile warning flags depending on the compiler
-ADJUST_METAL_COMPILER_WARNINGS()
-
-add_library(compiler_flags INTERFACE)
-target_link_libraries(
-    compiler_flags
-    INTERFACE
-        linker_flags
-        compiler_warnings
+string(TOUPPER "$ENV{ARCH_NAME}" ARCH_NAME_DEF)
+add_compile_definitions(ARCH_${ARCH_NAME_DEF})
+add_compile_options(
+    -Werror
+    -Wdelete-non-virtual-dtor
+    -Wreturn-type
+    -Wswitch
+    -Wuninitialized
+    -Wno-unused-parameter
+    -mavx2
+    -fPIC
+    -fvisibility-inlines-hidden
+    -fno-lto # FIXME: This seems to be here for ttnn; it should go to TTNN, then.
+    "$<$<CXX_COMPILER_ID:Clang>:-Wsometimes-uninitialized>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-c++11-narrowing>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-error=local-type-template-args>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-delete-non-abstract-non-virtual-dtor>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-c99-designator>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-shift-op-parentheses>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-non-c-typedef-for-linkage>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-this-capture>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-volatile>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-builtins>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-declarations>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-deprecated>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-deprecated-declarations>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-attributes>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overread>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overflow>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-maybe-uninitialized>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-missing-requires>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-narrowing>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-non-template-friend>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-error=non-template-friend>"
+    "$<$<BOOL:${ENABLE_ASAN}>:-fsanitize=address>"
+    "$<$<BOOL:${ENABLE_MSAN}>:-fsanitize=memory>"
+    "$<$<BOOL:${ENABLE_TSAN}>:-fsanitize=thread>"
+    "$<$<BOOL:${ENABLE_UBSAN}>:-fsanitize=undefined>"
```

Comment:
LETS FIX THE NEED FOR THESE WARNINGS IN THE C++!!! WHO IS READY TO VOLUNTEER!

Reply:
By "fix the need for these warnings" I think you mean "fix the need for _disabling_ these warnings"?